numerical_methods/secant_method.c

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file
+ * @brief Find a continuous function's root by using a succession 
```

Comment:
You really should include some kind of reference source. Here, I added a Wikipedia link for you
```suggestion
 * @brief [Wikipedia link](https://en.wikipedia.org/wiki/Secant_method). Find a continuous function's root by using a succession 
```